numpy/_core/include/numpy/dtype_api.h

```diff
@@ -477,4 +481,20 @@ typedef PyArray_Descr *(PyArrayDTypeMeta_FinalizeDescriptor)(PyArray_Descr *dtyp
 typedef int(PyArrayDTypeMeta_SetItem)(PyArray_Descr *, PyObject *, char *);
 typedef PyObject *(PyArrayDTypeMeta_GetItem)(PyArray_Descr *, char *);
 
+typedef int (PyArray_CompareFuncWithDescr)(const void *, const void *,
+                                           PyArray_Descr *);
+typedef int (PyArray_SortCompareFunc)(const void *, const void *,
+                                           PyArray_Descr *);                              
+typedef int (PyArray_SortFunc)(PyArrayMethod_Context *, 
+                               void *, npy_intp, 
+                               NpyAuxData *);
+typedef int (PyArray_ArgSortFunc)(PyArrayMethod_Context *, 
+                                  void *, npy_intp *, npy_intp, 
+                                  NpyAuxData *);
```

Comment:
These two need different names and you need to leave the original typedefs in `ndarraytypes.h` that had these names, since they're public API.

Reply:
Is it intended that a single sort function should handle all variations of the context, or is this basically a dispatch function?

Reply:
If dispatch, I would put that in the name: `PyArray_ArgSortDispatch`.